### DISCUSSION TITLE
Why build number or metadata is not available for task?

### DISCUSSION THREAD

**Discussion by Mizugorou** [concourse/concourse#6555] [Help & Support] [answered]
Hi,
I just newbie to concourse. I use Concourse to e2e pipeline with Gitlab. I want to send Concourse status back to Gitlab [commit api](https://docs.gitlab.com/ee/api/commits.html#post-the-build-status-to-a-commit), but I found  [stackover answer](https://stackoverflow.com/questions/39771347/concourse-ci-and-build-number/39775099?r=SearchResults&s=8|38.0807#39775099) that build number is not available for task. Could you tell why? or any suggest about this.

**Comment by taylorsilva** [comment, marked as answer]
Mainly this, from the docs https://concourse-ci.org/implementing-resource-types.html#resource-metadata

> They should not be used to emulate versioning (e.g. by using the increasing build number). They are not provided to task steps to avoid this anti-pattern.

Related discussion: https://github.com/concourse/concourse/discussions/6352

And of course there are ways around this, like having a resource pass in the metadata as an input: https://github.com/swce/metadata-resource

**Reply by Mizugorou** [reply]
Thank your answer

**Comment by gaigaslab-operations** [comment]
@taylorsilva  It appears that 7.0.0 breaks the metadata resource type.  See: https://github.com/concourse/concourse/discussions/6594
